ddtrace/contrib/trace_utils.py

```diff
@@ -577,7 +577,10 @@ def activate_distributed_headers(tracer, int_config=None, request_headers=None,
         #     app = Flask(__name__)  # Traced via Flask instrumentation
         #     app = DDWSGIMiddleware(app)  # Extra layer on top for WSGI
         current_context = tracer.current_trace_context()
-        if current_context and current_context.trace_id == context.trace_id:
+
+        # We accept incoming contexts with only baggage or only span_links, however if we
+        # already have a current_context then a baggage only or span_link only context will be tossed out
+        if current_context and (not context.trace_id or current_context.trace_id == context.trace_id):
```

Comment:
With the proposed logic 
if:
 - we have a current_context AND 
 - the extracted context does not have a trace_id 
then:
- we deactivate current_context and replace it with the new context

This feels a bit off. Instead I think we should:
- keep the current_context active AND
- add the baggage and span links in the new context to the current context. 

Activating a context with no trace_id will break the existing trace